pkg/plugin/cilium/cilium_linux.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+// Package cilium contains the Retina Cilium plugin. It uses unix socket to get events from cilium and decode them to flow objects.
+package cilium
+
+import (
+	"context"
+	"encoding/gob"
+	"errors"
+	"io"
+	"net"
+	"time"
+
+	v1 "github.com/cilium/cilium/pkg/hubble/api/v1"
+	"github.com/cilium/cilium/pkg/monitor/payload"
+	kcfg "github.com/microsoft/retina/pkg/config"
+	"github.com/microsoft/retina/pkg/enricher"
+	"github.com/microsoft/retina/pkg/log"
+	"github.com/microsoft/retina/pkg/plugin/api"
+	"go.uber.org/zap"
+)
+
+const (
+	MonitorSockPath1_2 = "/var/run/cilium/monitor1_2.sock"
+	defaultAttempts    = 10
+	defaultRetryDelay  = 12 * time.Second
+)
+
+func New(cfg *kcfg.Config) api.Plugin {
+	return &cilium{
+		cfg:         cfg,
+		l:           log.Logger().Named(string(Name)),
+		retryDelay:  defaultRetryDelay,
+		maxAttempts: defaultAttempts,
+		sockPath:    MonitorSockPath1_2,
+	}
+}
+
+func (c *cilium) Name() string {
+	return string(Name)
+}
+
+func (c *cilium) Generate(_ context.Context) error {
+	return nil
+}
+
+func (c *cilium) Compile(_ context.Context) error {
+	return nil
+}
+
+func (c *cilium) Init() error {
+	c.p = &parser{
+		l: c.l,
+	}
+	err := c.p.Init()
+	if err != nil {
+		c.l.Error("Failed to initialize cilium parser", zap.Error(err))
+		return err
+	}
+	c.l.Info("Initialized cilium plugin")
+	return nil
+}
+
+func (c *cilium) Start(ctx context.Context) error {
+	if c.cfg.EnablePodLevel {
+		if enricher.IsInitialized() {
+			c.enricher = enricher.Instance()
+		} else {
+			c.l.Warn("retina enricher is not initialized")
+		}
+	}
```

Comment:
If this plugin will be running with `hubble` controller and not `legacy` then I don't think this is required